src/pluggy/hooks.py

```diff
@@ -264,25 +264,16 @@ def _add_hookimpl(self, hookimpl):
                 i -= 1
             methods.insert(i + 1, hookimpl)
 
-        if "__multicall__" in hookimpl.argnames:
-            warnings.warn(
-                "Support for __multicall__ is now deprecated and will be"
-                "removed in an upcoming release.",
-                DeprecationWarning,
-            )
-            self.multicall = _legacymulticall
```

Comment:
Probably the `self.multicall` indirection is not needed anymore -- can call `_multicall` directly?

Reply:
@bluetech actually, come to think of this I wonder if it makes sense to keep this in preparation for something like #50? The overloading of the *multi-caller* I guess can be done anywhere?

I'd like @RonnyPfannschmidt's opinion on this.

Also this might bring up discussion on whether the `PluginManager._inner_hookexec` is something we could also get rid of since it only seems to be used for tracing iirc. Might be better addressed in a new issue/discussion tho.

Reply:
> actually, come to think of this I wonder if it makes sense to keep this in preparation for something like #50?

I'd say remove it and let #50 add it back, to show the "real cost", but up to you.

> Also this might bring up discussion on whether the PluginManager._inner_hookexec is something we could also get rid of since it only seems to be used for tracing iirc

pytest seems to use the tracing stuff (`add_hookcall_monitoring`)

Reply:
>pytest seems to use the tracing stuff 

Definitely, I've just wondered about implementing the tracing as part of the hook caller instead of some wrapper slapped in around the `_hookexec()` by the `PluginManager`.

Looking at this again I'm more convinced all that tracing magic should be implemented on the `_HookCaller` including adding a method to do the [`.subset_hook_caller()` stuff](https://github.com/pytest-dev/pluggy/blob/master/src/pluggy/manager.py#L362). In particular it's currently limiting us from doing *per-hook* tracing as well.

If you want to do tracing around the caller I'm not sure why that has anything to do (abstraction wise) with the plugin manager.

Reply:
@bluetech either way I dropped the `_HookCaller.multicall` here.

Reply:
Made #262 regarding my 2nd last comment.